lib/src/flutter_cli.dart

```diff
@@ -1,15 +1,66 @@
+import 'package:path/path.dart' as p;
 import 'package:universal_io/io.dart';
 
 /// Flutter CLI
 class Flutter {
   /// Install flutter dependencies (`flutter packages get`).
-  static Future<void> packagesGet([String? cwd]) {
-    return _Cmd.run('flutter', ['packages', 'get'], workingDirectory: cwd);
+  static Future<void> packagesGet({String? cwd, bool recursive = true}) async {
+    var foundPubspec = false;
+
+    if (recursive) {
+      final futures =
+          Directory(cwd ?? '').listSync(recursive: true).where((element) {
+        final e = element;
+        if (e is! File) return false;
+
+        return p.basename(e.path) == 'pubspec.yaml';
+      }).map(
+        (element) {
+          foundPubspec = true;
+          return _Cmd.run(
+            'flutter',
+            ['packages', 'get'],
+            workingDirectory: element.parent.path,
+          );
+        },
+      );
+
+      await Future.wait(futures);
+    }
+
+    if (foundPubspec == false) {
```

Comment:
If we didn't find a `pubspec` why would we run `flutter packages get`? I think we should invert the logic and if recursive is false just execute flutter packages get in the current working directory and return early.

Reply:
do we want an exception to be thrown when there is no pubspec.yaml? Because that logic is handled as part of the _Cmd class which is why I just call run on the cwd. I agree though it should be refactored.